src/evm-wallet-adapter/adapter.ts

```diff
@@ -12,38 +12,34 @@ export class CrossmintEVMWalletAdapter {
     icon = CROSSMINT_LOGO_21x21;
 
     private _connecting: boolean;
-    private _publicKeys: string[];
 
     private _config: CrossmintEmbedConfig;
     private _client?: CrossmintEmbed;
 
+    private _accounts?: (WalletProjection | EVMAAWalletProjection)[]
+
     constructor(
         params: Omit<CrossmintEmbedParams, "chain"> & { chain: BlockchainTypes.ETHEREUM | BlockchainTypes.POLYGON }
     ) {
         this._connecting = false;
-        this._publicKeys = [];
 
         this._config = buildConfig({ ...params });
     }
 
-    get publicKey(): string | null {
-        if (this._publicKeys.length == 0) return null;
-
-        return this._publicKeys[0];
+    get publicKey() {
+        return this._accounts?.[0].address;
```

Comment:
I don't think this dot is meant to be there?
```suggestion
        return this._accounts?[0].address;
```

Reply:
It's meant. This is the way optional chaining works when dealing with arrays